.bazelrc

```diff
@@ -36,6 +36,9 @@ build --action_env=CXX
 build --action_env=LLVM_CONFIG
 build --action_env=PATH
 
+# Skip system ICU linking.
+build --@com_googlesource_googleurl//build_config:system_icu=0
```

Comment:
Is there any way in CI to verify that we definitely don't have ICU in the binary somewhow? E.g. with readelf or the like.

Reply:
Added.

Reply:
I think we may not need to deal with ICU at all here. We just need the `ParseStandardURL` function which does use or include anything from ICU.
I think what would work is this:
1. define new cc_library in the googleurl/url/BUILD which only includes sources necessary to build the `ParseStandardURL` function. I think this should not involve any files that reference ICU.

Note that we may also need the `RemoveURLWhitespace` function. I'm yet unsure if this function is needed. I'll look into it later tonight.
1. Add flag protected change to the `Http::Utility::Url` class to use the `ParseStandardURL` from google URL.
1. Make sure all existing tests pass.

Reply:
@yanavlasov sorry, a question. I can definitely pull only the `ParseStandardURL` related files (that will be `url/third_party/mozilla/url_parse.{cc, h}`) and let the `Http::Utility::Url` implementation to use that, however, how about: https://github.com/envoyproxy/envoy/blob/deded5308058e1484f36779b0a4f7bab0fc009a8/source/common/http/path_utility.cc#L3-L4? Are we going to keep `chromium_url` around? cc. @htuch.

Reply:
We will removal `common/chromium_url` in favor of this eventually, but will need to keep it around while we do the runtime guarded transition.

We should still verify ICU is not creeping into the build, since this is a regression check for the previous CVE.

Reply:
As @htuch  said we will not keep Envoy's fork of the Chromium URL in `common/chromium_url` around. We will use the same files from the external chromium URL package you are adding now. We can deal it with in a follow up PR for simplicity.

Reply:
OK. I have added a check on ICU deps https://github.com/envoyproxy/envoy/pull/13808/files#diff-2d449e3aa6794c338469e0083b7fcd58ed4056cab7efda61e1989060f59b682cR12 and run unit tests for the static linked ICU. 

If it is OK, can we land this (since it is not used anywhere other than tests for itself) and do necessary on replacing chromium_url afterward? 

Anything else I need to add?

Reply:
Most of the functionality that this PR brings will not be used by Envoy. For example we will not use GURL class. I'm also unsure of the value of fuzzing it, since this done by Chrome already.
I think this PR could be made quite a bit smaller if we compile just the files that are needed by the `ParseStandardURL` which is what will be needed in Envoy. I think ICU shim can be done away with in this case. So I would vote for reworking this PR as we have discussed above unless there is another compelling reason to land this.

Reply:
Sorry that I haven't got a clear signal. @htuch to make it clear, are we going with @yanavlasov suggestion above or to land this PR? Thanks!

Reply:
Ultimately we want something to replace the uses in https://github.com/envoyproxy/envoy/tree/master/source/common/chromium_url, which are around `url::CanonicalizePath()`. Can we do this without the ICU shim? What about pulling other URL parsing functionality out? Do we need build gymnastics to skip ICU in these cases, e.g. patch files?

Reply:
OK. So from my understanding, the caller site is https://github.com/envoyproxy/envoy/blob/dd0befa3e1e55e408329c7fb7356a84462416b02/source/common/http/path_utility.cc#L15-L28 plus we need `ParseStandardURL`. I will try to look at this.

Reply:
After a few attempts, I think we can't do a clean "import" e.g. referencing `third_party/mozilla/url_parse.cc`  from a BUILD file in envoy can cause a visibility problem (can be solved by using `exports_files` but that requires a patch to upstream), without as @htuch said, a custom genrule_cmd.

re: fuzzing, the motivation was to make sure if we have a googleurl lib linked with the shimmed ICU lib should not causing crashes (but yeah this should be predictable by inputting URLs with IDN to the function). If we want to skip testing it via GURL class, I can remove the test and do testing by calling "url::CanonicalizePath" instead.

Reply:
I think a simple `exports_files` patch is OK. What we're really weary of is complex patches that are likely to break on version upgrades.

Reply:
I see. Submitted a draft here: https://github.com/envoyproxy/envoy/pull/14583, patching the googleurl's url/BUILD file.